frontend/src/components/CommentForm.tsx

```diff
@@ -0,0 +1,112 @@
+// CommentForm.tsx
+import React, { useState } from "react";
+import {
+  Box,
+  FormControl,
+  FormLabel,
+  Textarea,
+  Button,
+  useToast,
+} from "@chakra-ui/react";
+import { Comment, postComment } from "../hooks/usePublicDecks";
+import { useUserContext } from "./UserContext";
+
+interface CommentFormProps {
+  deckId: number;
+  // Remove onCommentAdded prop since it's no longer needed
+  // onCommentAdded: (comment: Comment) => void;
+}
+
+const CommentForm: React.FC<CommentFormProps> = ({ deckId }) => {
+  const [commentText, setCommentText] = useState("");
+  const { token, username } = useUserContext();
+  const toast = useToast();
+
+  if (!token || !username) {
+    return null;
+  }
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!commentText.trim()) {
+      toast({
+        title: "Error",
+        description: "Comment cannot be empty.",
+        status: "error",
+        duration: 5000,
+        isClosable: true,
+      });
+      return;
+    }
+
+    const newComment: Comment = {
+      text: commentText,
+      username: username,
+    };
+
+    try {
+      const response = await postComment(deckId, newComment);
+      if (response.data) {
+        setCommentText("");
+
+        // Show success toast
+        toast({
+          title: "Comment Posted",
+          description: "Your comment has been posted successfully.",
+          status: "success",
+          duration: 5000,
```

Comment:
For at have det ensformigt i toasts vil jeg anbefale at vi ikke definerer længden den skal være åben, da det giver en masse ekstra linjer man skal skrive, så plz fjern den del